tests/test_integrators.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+import aesara
+import aesara.tensor as aet
+import numpy as np
+from aesara.tensor.var import TensorVariable
+
+from aesara_hmc.integrators import velocity_verlet
+
+
+def HarmonicOscillator(inverse_mass_matrix, k=5, m=1.0):
+    """Potential and Kinetic energy of an harmonic oscillator."""
+
+    def potential_energy(x: TensorVariable) -> TensorVariable:
+        return aet.sum(0.5 * k * aet.square(x))
+
+    def kinetic_energy(p: TensorVariable) -> TensorVariable:
+        v = inverse_mass_matrix * p
+        return aet.sum(0.5 * aet.dot(v, p))
+
+    return potential_energy, kinetic_energy
+
+
+def FreeFall(inverse_mass_matrix, g=9.81, m=1.0):
+    """Potential and kinetic energy of a free-falling object."""
+
+    def potential_energy(h: TensorVariable) -> TensorVariable:
+        return aet.sum(m * g * h)
+
+    def kinetic_energy(p: TensorVariable) -> TensorVariable:
+        v = inverse_mass_matrix * p
+        return aet.sum(0.5 * aet.dot(v, p))
+
+    return potential_energy, kinetic_energy
+
+
+integration_examples = [
+    {
+        "model": HarmonicOscillator,
+        "n_steps": 100,
+        "step_size": 0.01,
+        "q_init": np.array([0.0]),
+        "p_init": np.array([1.0]),
+        "inverse_mass_matrix": np.array([1.0]),
+    },
+    {
+        "model": FreeFall,
+        "n_steps": 100,
+        "step_size": 0.01,
+        "q_init": np.array([0.0]),
+        "p_init": np.array([1.0]),
+        "inverse_mass_matrix": np.array([1.0]),
+    },
+]
+
+
+def integrate_fn(potential, step_fn, n_steps):
+    q = aet.vector("q")
+    p = aet.vector("p")
+    step_size = aet.scalar("step_size")
+    energy = potential(q)
+    energy_grad = aesara.grad(energy, q)
+    trajectory, _ = aesara.scan(
+        fn=step_fn,
+        outputs_info=[q, p, energy, energy_grad],
+        non_sequences=step_size,
+        n_steps=n_steps,
+    )
```

Comment:
This fails with the following error message:

```
ValueError: Scan cannot parse the return value of your lambda expression, which is: (Elemwise{add,no_inplace}.0, Elemwise{sub,no_inplace}.0, Sum{acc_dtype=float64}.0, [Elemwise{mul}.0])
```

and I cannot figure out why after reading the documentation and the code.